guides/source/association_basics.md

```diff
@@ -355,2288 +447,2474 @@ class CreateAccountHistories < ActiveRecord::Migration[7.2]
+section](#has-many-through-vs-has-and-belongs-to-many).
 
-class Chapter < ApplicationRecord
-  belongs_to :book
-end
-```
+#### Creating Foreign Keys for `belongs_to` Associations
 
-If you frequently retrieve chapters directly from authors (`@author.books.chapters`), then you can make your code somewhat more efficient by including chapters in the association from authors to books:
+When you declare a [`belongs_to` association](#belongs-to), you need to create
+foreign keys as appropriate. For example, consider this model:
 
 ```ruby
-class Author < ApplicationRecord
-  has_many :books, -> { includes :chapters }
-end
-
 class Book < ApplicationRecord
   belongs_to :author
-  has_many :chapters
 end
+```
+
+This declaration needs to be backed up by a corresponding foreign key column in
+the books table. For a brand new table, the migration might look something like
+this:
 
-class Chapter < ApplicationRecord
-  belongs_to :book
+```ruby
+class CreateBooks < ActiveRecord::Migration[7.2]
+  def change
+    create_table :books do |t|
+      t.datetime   :published_at
+      t.string     :book_number
+      t.belongs_to :author
+    end
+  end
 end
 ```
 
-##### `limit`
-
-The `limit` method lets you restrict the total number of objects that will be fetched through an association.
+Whereas for an existing table, it might look like this:
 
 ```ruby
-class Author < ApplicationRecord
-  has_many :recent_books,
-    -> { order('published_at desc').limit(100) },
-    class_name: "Book"
+class AddAuthorToBooks < ActiveRecord::Migration[7.2]
+  def change
+    add_reference :books, :author
+  end
 end
 ```
 
-##### `offset`
 
-The `offset` method lets you specify the starting offset for fetching objects via an association. For example, `-> { offset(11) }` will skip the first 11 records.
+#### Creating Join Tables for `has_and_belongs_to_many` Associations
+
+If you create a `has_and_belongs_to_many` association, you need to explicitly
+create the join table. Unless the name of the join table is explicitly specified
+by using the `:join_table` option, Active Record creates the name by using the
+lexical order of the class names. So a join between author and book models will
+give the default join table name of "authors_books" because "a" outranks "b" in
+lexical ordering.
 
-##### `order`
+WARNING: The precedence between model names is calculated using the `<=>`
+operator for `String`. This means that if the strings are of different lengths,
+and the strings are equal when compared up to the shortest length, then the
+longer string is considered of higher lexical precedence than the shorter one.
+For example, one would expect the tables "paper_boxes" and "papers" to generate
+a join table name of "papers_paper_boxes" because of the length of the name
+"paper_boxes", but it in fact generates a join table name of
+"paper_boxes_papers" (because the underscore '\_' is lexicographically _less_
+than 's' in common encodings).
```

Comment:
This warning probably makes more sense in the migrations guide.

Reply:
Agreed its too much detail for here - Moved it to the relevant section in migrations.